exploration/0001-builtin-registry-capabilities.md

```diff
@@ -16,31 +16,35 @@ Status: **Proposed**
+This document aims to list down the basic capabilities that need to be provided to developers by a _built-in registry_.
 
 ## Use-Cases
 
-From formatting date and time values to performing selection based on plural rules, the use cases for a minimal set of common tools standardized across implementations are virtually everywhere. While a number of applications might require more complex functions with custom logic, a decently designed set of built-in functions could significantly lower the barrier of entry as well as the need to write many custom functions.
+From formatting date and time values to performing selection based on plural rules, the use cases for a minimal set of common tools standardized across implementations are virtually everywhere.
+While a number of applications might require more complex functions with custom logic, a decently designed set of _built-in functions_ could significantly lower the barrier of entry as well as the need to write and maintain many custom functions.
 
 ## Requirements
 
-Implementations are not required to implement and expose all the built-in functions that would result from this initiative. However, they are required to strictly stick to the standardized interface for the functions they do expose.
+Implementations are required to implement and expose the entire _built-in registry_ that would provide the capabilities in this document.
 
 ## Constraints
 
-Like many standardized APIs out there, the biggest constraint on the resulting set of built-in functions would be that until we decide to make another major update to MessageFormat, we'd not be able to make breaking changes to the function interfaces since that would invalidate the promise of a consistent API across implementations and versions.
+Like many standardized APIs out there, the biggest constraint on the resulting set of _built-in functions_ would be that until we decide to make another major update to MessageFormat, we'd not be able to make breaking changes to the function signatures since that would invalidate the promise of a consistent API across implementations and versions.
 
 Another constraint is that since MessageFormat would be used alongside other internationalization features in many programming environments, the final interface should be somewhat consistent to reduce developer confusion.
 
 ## Proposed Design
 
-All conforming implementations of this specification MUST provide a set of <dfn>built-in functions</dfn>, which allows the developers to perform the following operations:
+The **_<dfn>built-in registry</dfn>_** is the set of _functions_, together with their options, that are present in every implementation.
+Implementations MUST provide each of the built-in functions defined in the _built-in registry_, including all of the options.
+They MAY include additional options, although implementers are cautioned to avoid options which might conflict with future standardization.
```

Comment:
This is fine, but could include a clarification (potentially as a later change) that the exact output produced for any of these functions or options is left up to each implementation.

So for example a conformant implementation could "support" the `notation` option for `:number` by not throwing an error if it's used, but producing the same output no matter what the value the option has.